internal/repository/sync.go

```diff
@@ -64,8 +68,33 @@ func Clone(ctx context.Context, remoteURL, dir, initialBranch string) (*Reposito
 		return nil, errors.Join(ErrCloningRepository, err)
 	}
 
+	sort.Slice(expectedRootKeys, func(i, j int) bool {
+		return expectedRootKeys[i].KeyID < expectedRootKeys[j].KeyID
+	})
+
 	repository := &Repository{r: r}
 
 	slog.Debug("Verifying HEAD...")
+
+	if len(expectedRootKeys) > 0 {
+		state, err := policy.LoadCurrentState(ctx, r)
```

Comment:
To be clear, I think this should be the _first_ ever complete policy state. This feature is a mechanism to bootstrap the trust for the first ever* root.json. Subsequent versions of root (including their keys) can be verified using the immediate preceding version of the root. It's only the very first root keys we need to bootstrap entirely independently.

\* That said, there may be situations where there's been a compromise of a threshold of root keys, but we'll have to account for checkpointing that separately anyway.

Reply:
I have fixed this